testcontainers/src/core/image/image_ext.rs

```diff
@@ -169,6 +169,12 @@ pub trait ImageExt<I: Image> {
 
     /// Sets the user that commands are run as inside the container.
     fn with_user(self, user: impl Into<String>) -> ContainerRequest<I>;
+
+    /// Sets the container's root filesystem to be mounted as read-only
+    fn with_readonly_rootfs(self, readonly_rootfs: Option<bool>) -> ContainerRequest<I>;
+
+    /// Sets security options for the container
+    fn with_security_opt(self, security_opt: Option<Vec<String>>) -> ContainerRequest<I>;
```

Comment:
Also, it would be nice to keep aligned "append" semantic  for `security_opt` as we do for capabilities for example

1. Accept `opt: impl Into<String>`
2. Append to the list
```
        container_req
            .security_opts
            .get_or_insert_with(Vec::new)
            .push(capability.into());
```

Reply:
This also may prevent us from having `Some( vec![] )` (empty vec inside the Some)

Reply:
Fixed